encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
```

Comment:
I'm not sure where does this '10' come from. Should be an infinite-ish loop here. A large upload can take potentially many hours.

Reply:
infinite-ish is almost infinite so switched to `while True`